task-sdk/src/airflow/sdk/log.py

```diff
@@ -96,6 +96,13 @@ def redact_jwt(logger: Any, method_name: str, event_dict: EventDict) -> EventDic
     return event_dict
 
 
+def mask_logs(logger: Any, method_name: str, event_dict: EventDict) -> EventDict:
+    from airflow.sdk.execution_time.secrets_masker import redact
+
+    event_dict["event"] = redact(event_dict["event"])
+    return event_dict
```

Comment:
Could you add a simple test case for it

Reply:
We need to look at pretty much every field in `event_dict`:

If I do this:

```python
log.debug("Connecting", password=pass)`
```

it won't get redacted.

You can just pass the whole dict to redact I think, and it will then also use the dict key names to redact, not just values.

However I think you will run into issues with this fn https://github.com/apache/airflow/blob/main/task-sdk/src/airflow/sdk/execution_time/secrets_masker.py#L192-L210 which expects a stdlib LogRecord object which we don't have anymore.

Reply:
Thanks for the clarification, let me try that out.

Reply:
Added a test and did the needful!